pkg/collectors/devices/device_info.go

```diff
@@ -106,20 +113,34 @@ func processGNSSPath(s string) (string, error) {
 	return "/dev/" + strings.TrimSpace(s), nil
 }
 
-// BuildPTPDeviceInfo popluates the fetcher required for
-// collecting the PTPDeviceInfo
-func BuildPTPDeviceInfo(interfaceName string) error { //nolint:dupl // Further dedup risks be too abstract or fragile
+func getGNSSSCommand(ctx clients.ExecContext, interfaceName string) (*clients.Cmd, error) {
+	buf := bytes.Buffer{}
+	buf.WriteString(fmt.Sprintf("ls /sys/class/net/%s/device/gnss/", interfaceName))
+	stdout, _, err := ctx.ExecCommandStdIn([]string{"/usr/bin/sh"}, buf)
+	if err != nil || stdout == "" {
+		return nil, fmt.Errorf("command to find gnss devices: %w", err)
+	}
```

Comment:
I'm confused about the flow here - We execute this ls command, and if it succeeds we discard the result and then create a command that will run it again?

Also we're duplicating the same `fmt.Sprintf(...)` (minor nit)

Reply:
Ah, maybe because the command we return is the one that gets executed in a way that doesn't allow for errors?

Reply:
Yeah the fetchers are less tolerant of errors than running it as a normal exec.